### PR TITLE
Fix-up some AWS Auth Chain issues

### PR DESCRIPTION
**Pull request by mmathias01**
Hey all!

I was deploying a new pipeline the other day in Kubernetes and ran in to this issue: [Snowplow Collector on K8S does not use service account
](https://stackoverflow.com/questions/67175006/snowplow-collector-on-k8s-does-not-use-service-account)

After some extensive debugging I was able to determine that the reason why is because in order to use the IAM role assigned to ServiceAccount you must have the STS jar in your classpath. 

This pull request resolves that issue and allows the container to correctly use the IAM role associated with a ServiceAccount. 

Hopefully this helps. For the time being I am using a container image I built myself, but figured I may as well let you know what the fix to this issue was!

### CODE REVIEW

File: .jvmopts [hunk 1]
```diff
@@ -8,5 +8,5 @@
 -XX:+TieredCompilation
 -XX:-UseGCOverheadLimit
 # effectively adds GC to Perm space
--XX:+CMSClassUnloadingEnabled
+#-XX:+CMSClassUnloadingEnabled
```
Comment (mmathias01) [inline .jvmopts:11]: I had to do this in order to get it to build on my system. According to the interwebs this flag might have been deprecated in new JVMs? This is obviously not required as part of the fix.

File: build.sbt [hunk 2]
```diff
@@ -75,6 +75,7 @@ lazy val allStreamSettings = BuildSettings.basicSettings ++ BuildSettings.sbtAss
     Dependencies.Libraries.slf4j,
     Dependencies.Libraries.log4jOverSlf4j,
     Dependencies.Libraries.s3Sdk,
+    Dependencies.Libraries.sts,
```
Comment (mmathias01) [inline build.sbt:78]: included here since s3 is included here, so IAM roles could be useful.

File: project/Dependencies.scala [hunk 3]
```diff
@@ -58,7 +58,7 @@ object Dependencies {
     val collectorPayload = "0.0.0"
     val schemaSniffer    = "0.0.0"
 
-    val awsSdk           = "1.11.728"
+    val awsSdk           = "1.11.822"
```
Comment (mmathias01) [inline project/Dependencies.scala:61]: bumped version to match other components in the pipeline (collector, etc.)

File: project/Dependencies.scala [hunk 4]
```diff
@@ -151,6 +151,7 @@ object Dependencies {
     val dynamodbSdk      = "com.amazonaws"                    %  "aws-java-sdk-dynamodb"             % V.awsSdk
     val s3Sdk            = "com.amazonaws"                    %  "aws-java-sdk-s3"                   % V.awsSdk
     val kinesisClient    = "com.amazonaws"                    %  "amazon-kinesis-client"             % V.kinesisClient
+    val sts              = "com.amazonaws"                    %  "aws-java-sdk-sts"                  % V.awsSdk
```
Comment (mmathias01) [inline project/Dependencies.scala:154]: added row to dependencies here.

**Comment by mmathias01** [comment]
For more information see [this comment](https://github.com/snowplow/stream-collector/pull/160#discussion_r704797877) on the collector PR

**Comment by mmathias01** [comment]
@istreeter Just making you aware that this is here, although I am sure you know already. This is the same thing as we discussed on the collector repo!

**Comment by istreeter** [comment]
Hi @mmathias01 -- @benjben knows about adding sts as a runtime dependency, and he will be including it the release which is currently in preparation.

**Comment by benjben** [comment]
Thanks a lot for the contribution @mmathias01 ! I'll merge it into the branch where I'm currently working on

**Comment by benjben** [comment]
I can't rebase and merge as you branched off from a different commit than mine, so I'll copy your changes and set you as the author

**Comment by benjben** [comment]
Handle in #484

**Pull request closed without merging**